tests/e2e/testsLibrary/che-code/ProjectAndFileTestsCheCode.ts

```diff
@@ -10,24 +10,103 @@
 
 import 'reflect-metadata';
 import { inject, injectable } from 'inversify';
-import { By, until } from 'selenium-webdriver';
+import { By, error, until, WebElement } from 'selenium-webdriver';
 import { DriverHelper } from '../../utils/DriverHelper';
 import { CLASSES } from '../../inversify.types';
 import { Logger } from '../../utils/Logger';
 import { TimeoutConstants } from '../../TimeoutConstants';
+import { Workbench, SideBarView, ViewContent, ViewSection, ViewItem } from 'monaco-page-objects';
+import { TestConstants } from '../../TestConstants';
 
 @injectable()
 export class ProjectAndFileTestsCheCode {
 
+    private static readonly ENDPOINTS_TITLE_TEXT: string = 'endpoints';
+    private static readonly TRUST_AUTHORS_DIALOG_BOX_TEXT: string = 'Do you trust the authors of the files in this folder?';
+    private static readonly TRUST_AUTHORS_DIALOG_BUTTON_TEXT: string = 'Yes, I trust the authors';
+
     constructor(
         @inject(CLASSES.DriverHelper) private readonly driverHelper: DriverHelper) {}
 
-    public async waitWorkspaceReadinessForCheCodeEditor(): Promise<void> {
+    public async waitWorkspaceReadinessForCheCodeEditor(): Promise<Workbench> {
```

Comment:
It's important to make sure the method supports `TS_SELENIUM_START_WORKSPACE_TIMEOUT` completely, to avoid the issue is reproduced again: https://github.com/eclipse/che/issues/21965